amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/contexts/WebApiContext.scala

```diff
@@ -122,20 +124,13 @@ abstract class WebApiContext(val loc: String,
 
   def computeJsonSchemaVersion(ast: YNode): SchemaVersion = parseSchemaVersion(ast, eh)
 
-  def resolvedPath(base: String, str: String): String =
-    if (str.isEmpty) platform.normalizePath(base)
-    else if (str.startsWith("/")) str
-    else if (str.contains(":")) str
-    else if (str.startsWith("#")) base.split("#").head + str
-    else if (shouldSkipDot(base, str)) basePath(base) + str.substring(2)
-    else platform.normalizePath(basePath(base).urlDecoded + str)
-
-  def shouldSkipDot(base: String, path: String): Boolean =
-    (basePath(base) == "file://" || basePath(base) == "file:///") && path.startsWith("./")
-
-  def basePath(path: String): String = {
-    val withoutHash = if (path.contains("#")) path.split("#").head else path
-    withoutHash.splitAt(withoutHash.lastIndexOf("/"))._1 + "/"
+  def resolvedPath(current: String, url: String): String = {
```

Comment:
Should this be in another class and be unit tested separately?

Reply:
done.